### ISSUE TITLE
[Feature]:  parallel sampler

### ISSUE THREAD

**Issue by qlibp** [qlibp/vllm-ascend#27]
### 🚀 The feature, motivation and pitch

设计一个高性能的并行采样器（Parallel Sampler），支持在单次推理批次中同时对 n 个独立序列进行采样，并灵活应用 Temperature、Top-K、Top-P、Min-P 等采样参数，最小化采样延迟。
sampling_params = {
    "n": 5,                    # 生成5个独立序列
    "temperature": 0.8,         # 控制随机性
    "top_k": 50,               # Top-K采样：只保留概率最高的50个token
    "top_p": 0.9,              # Top-P采样（nucleus采样）
}

当 sampling 设置 n =K 时，prompt会先复制k次，或在第一次进行k次采样，生成k个独立的序列。
然后，在模型完成一次前向传播，为整个批次计算出 logits 后，采样器（Sampler）会一次性处理所有 K 个序列。它会向量化地应用温度缩放（temperature）、Top-K、Top-P 等参数，然后通过一次 torch.multinomial 操作或 argmax（在 temperature=0 时）为批次中的所有序列同时采样出下一个 token。
